packages/transformers/babel/src/babel7.js

```diff
@@ -8,6 +8,8 @@ import {relativeUrl} from '@parcel/utils';
 import {remapAstLocations} from './remapAstLocations';
 
 import packageJson from '../package.json';
+import {applicationProfiler} from '@parcel/profiler';
```

Comment:
I think it be better to have this be provided by core (so "dependency injection" similar to the `logger` right now).

This would break if Yarn doesn't hoist the package and then the calls `applicationProfiler.onTrace()` and `applicationProfiler.createMeasurement` would actually go to different instances.

We had this problem in the past: https://github.com/parcel-bundler/parcel/pull/6566

Reply:
Yep, that's fair enough - you mean just where it's used in plugins as opposed to core? 

FWIW the initial implementation that Brian did internally passed around a "tracer" through core as well - this worked okay for the top level metrics, but once you wanted to instrument anything deeper it was a real pain to thread it through all the various layers.

Reply:
The problem occurs if some package X has state (such as that profiler) and X is declared as a dependency in multiple packages. 
So just inside of core would be fine (as there's only one dependency and no way Yarn creates two instances), though if it's only used inside of core anyway then it could maybe also just be separate file in core (as opposed to a separate package).

Reply:
> you mean [using dependency injection] just where it's used in plugins as opposed to core?

Yes, in the public API